seastar/src/config_and_start_seastar.cc

```diff
@@ -0,0 +1,110 @@
+#include "config_and_start_seastar.hh"
+
+namespace seastar {
+
+// Copies rust::Vec<rust::String> as char**. Returns nullptr if fails.
+// Function free_args should be called on returned pointer to avoid memory leak.
+static char** args_as_ptr(const rust::Vec<rust::String>& args) {
+    char** av = (char**) calloc(args.size() + 1, sizeof(char*));
+    if (av == nullptr) {
+        return nullptr;
+    }
+
+    for (size_t i = 0; i < args.size(); i++) {
+        const rust::String& arg = args[i];
+        av[i] = (char*) calloc(arg.size() + 1, sizeof(char));
+        if (av[i] == nullptr) {
+            return nullptr;
+        }
+        strncpy(av[i], arg.data(), arg.size());
+    }
+
+    return av;
+}
+
+static void free_args(char** av, size_t ac) {
+    for (size_t i = 0; i < ac; i++) {
+        free(av[i]);
+    }
+    free(av);
+}
+
+std::unique_ptr<seastar_options> new_options() {
+    return std::make_unique<seastar_options>();
+}
+
+rust::Str get_name(const std::unique_ptr<seastar_options>& opts) {
+    return rust::Str(&*opts->name.begin(), opts->name.size());
+}
+
+rust::Str get_description(const std::unique_ptr<seastar_options>& opts) {
+    return rust::Str(&*opts->description.begin(), opts->description.size());
+}
+
+uint32_t get_smp(const std::unique_ptr<seastar_options>& opts) {
+    if (opts->smp_opts.smp) {
+        return (uint32_t)opts->smp_opts.smp.get_value();
+    } else {
+        return (uint32_t)get_current_cpuset().size();
+    }
+}
+
+void set_name(const std::unique_ptr<seastar_options>& opts, const rust::Str name) {
+    opts->name = sstring(name.begin(), name.size());
+}
+
+void set_description(const std::unique_ptr<seastar_options>& opts, const rust::Str description) {
+    opts->description = std::string(description.begin(), description.size());;
```

Comment:
Nit: double semicolon at the end of the line

Reply:
Must have been my Ocaml muscle memory. I just noticed, there should be `seastar::sstring`, not `std::string` too.